tests/integration_tests/functional/test_vsock.py

```diff
@@ -215,3 +218,93 @@ def test_vsock_transport_reset(
     check_host_connections(path, blob_path, blob_hash)
     metrics = vm2.flush_metrics()
     validate_fc_metrics(metrics)
+
+
+def test_vsock_transport_reset_g2h(
+    uvm_nano, microvm_factory, bin_vsock_path, test_fc_session_root_path
+):
+    """
+    Vsock transport reset test.
+    """
+    test_vm = uvm_nano
+    test_vm.add_net_iface()
+    test_vm.api.vsock.put(vsock_id="vsock0", guest_cid=3, uds_path=f"/{VSOCK_UDS_PATH}")
+    test_vm.start()
+    test_vm.wait_for_up()
+
+    # Generate the random data blob file.
+    blob_path, blob_hash = make_blob(test_fc_session_root_path)
+    vm_blob_path = "/tmp/vsock/test.blob"
+
+    # Set up a tmpfs drive on the guest, so we can copy the blob there.
+    # Guest-initiated connections (echo workers) will use this blob.
+    _copy_vsock_data_to_guest(test_vm.ssh, blob_path, vm_blob_path, bin_vsock_path)
```

Comment:
does this test need this?

Reply:
I tried to make the test look like a previous one. I don't mind removing it.

Reply:
I think the existing test is already very bad because it tests 3 different things (connection reset, metrics well-formedness and device works after snapshot), so I'd prefer to not replicate that (and I'd also love for the existing test to be split apart, but that's out of scope here).

Reply:
I cannot disagree here.

Reply:
removed